src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -214,31 +397,368 @@ public void log(Logger logger) {
+        public void setReadIgnoreUsers(Set<String> readIgnoreUsers) {
+            if (readIgnoreUsers != null) {
+                this.readIgnoreUsers = readIgnoreUsers;
+            } else {
+                this.readIgnoreUsers = Collections.emptySet();
+            }
         }
+
+        /**
+         * Checks if only metadata or field names are to be logged for doc write requests
+         * @return true/false
+         */
+        @JsonIgnore
+        public boolean isWriteMetadataOnly() {
+            return writeMetadataOnly;
+        }
+
+        /**
+         * Set whether to log only metadata for doc write requests
+         * @param writeMetadataOnly true/false
+         */
+        @JsonIgnore
+        public void setWriteMetadataOnly(boolean writeMetadataOnly) {
+            this.writeMetadataOnly = writeMetadataOnly;
+        }
+
+        /**
+         * Checks if only diffs are to be logged for doc write requests
+         * @return true/false
+         */
+        @JsonIgnore
+        public boolean isWriteLogDiffs() {
+            return writeLogDiffs;
+        }
+
+        /**
+         * Set whether to log only diffs for doc write requests
+         * @param writeLogDiffs true/false
+         */
+        @JsonIgnore
+        public void setWriteLogDiffs(boolean writeLogDiffs) {
+            this.writeLogDiffs = writeLogDiffs;
+        }
+
+        /**
+         * Get list of indices to watch for doc write requests
+         * @return indices
+         */
+        @JsonIgnore
+        public List<String> getWriteWatchedIndices() {
+            return writeWatchedIndices;
+        }
+
+        /**
+         * Set indices to watch for doc write requests
+         * @param writeWatchedIndices indices
+         */
+        @JsonSetter(value = Key.WRITE_WATCHED_INDICES, nulls = Nulls.AS_EMPTY)
+        public void setWriteWatchedIndices(List<String> writeWatchedIndices) {
+            if (writeWatchedIndices != null) {
+                this.writeWatchedIndices = writeWatchedIndices;
+            } else {
+                this.writeWatchedIndices = Collections.emptyList();
+            }
+        }
+
+        /**
+         * Get users ignored for auditing doc write requests
+         * @return set of ignored users
+         */
+        @JsonIgnore
+        public Set<String> getWriteIgnoreUsers() {
+            return writeIgnoreUsers;
+        }
+
+        /**
+         * Set users to be ignored for auditing doc write requests
+         * @param writeIgnoreUsers users
+         */
+        @JsonSetter(value = Key.WRITE_IGNORE_USERS, nulls = Nulls.AS_EMPTY)
+        public void setWriteIgnoreUsers(Set<String> writeIgnoreUsers) {
+            if (writeIgnoreUsers != null) {
+                this.writeIgnoreUsers = writeIgnoreUsers;
+            } else {
+                this.readIgnoreUsers = Collections.emptySet();
+            }
+        }
+    }
+
+    public static class Key {
```

Comment:
IMO, it will be better to move those strings to `Filter` or `Compliance` classes instead of sharing them in a single class. It is OK to do this in a follow up PR.

Reply:
Done!